### DISCUSSION TITLE
Brew uninstall --zap does not remove any files

### DISCUSSION THREAD

**Discussion by SoCuul** [Homebrew/discussions#5790] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: e78a0adb4f260913fffdc956cc8c20fb54109feb
Last commit: 11 days ago
Branch: stable
Core tap HEAD: 414f5d990b76352b172bbe57f3c67a8ad5376add
Core tap last commit: 63 minutes ago
Core tap JSON: 30 Nov 09:45 UTC
Core cask tap HEAD: 1ef784f287cce93f2bb54bea66ae7ac0953f623b
Core cask tap last commit: 6 days ago
Core cask tap JSON: 30 Nov 09:45 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.yMDGr0ARyL/org.xquartz:0
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_NO_INSTALL_FROM_API: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: 16-core 64-bit dunno
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1.1-arm64
CLT: 16.0.0.26.4
Xcode: 16.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  cowsay

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libfuse-t-1.0.44.dylib

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/fuse-t.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libfuse-t-1.0.44.a
```

### Description of issue

Whenever I run `brew uninstall --zap <app>` the configuration files are not removed.

The files that should be removed are displayed in the CLI, but after the uninstall is finished, none of the files mentioned have been touched.

### Relevant casks

All

**Comment by SoCuul** [comment]
I had previously posted this issue here, but it was falsely closed with the issue not being resolved.

**Comment by SMillerDev** [comment]
> Whenever I run `brew uninstall --zap <app>` the configuration files are not removed.

There is no cask named `<app>`, if you want help you'll need to come up with examples that people can reproduce.

**Reply by SoCuul** [reply]
This has happened for me with every single cask that I have tried. An example provided here is Inkscape (GIMP is shown in the video but there happened to be other problems with its formula).

https://github.com/user-attachments/assets/dad657bd-053b-4e50-be12-e190d4bd90de

**Reply by gromgit** [reply]
If you're referring to [this discussion](https://github.com/orgs/Homebrew/discussions/5740), I'll note that you seem to have posted the same video you did the last time, which predated a PR to the GIMP cask that fixed incorrect file paths.

You refused to try that fix, which meant no progress could be made on your issue. That was probably why the previous discussion was closed.

**Reply by SoCuul** [reply]
If you read my comment on the previous post, you would have seen that:
> Well considering inkscape had the correct path the whole time and didn't work, nothing will have changed.

And it does in fact still happen with the latest version of the gimp cask.

**Reply by daeho-ro** [reply]
I know this is not helpful, but it works on my local... for `gimp`.

**Reply by SMillerDev** [reply]
Are you running brew from the same user that's using the cask?

**Reply by SoCuul** [reply]
I'm running brew from the same user as the cask.

To hopefully provide a little bit more insight, when I run `brew uninstall --cask --zap gimp; ls -l ~/Library/Preferences/org.gimp.gimp-2.10.plist` I get the following output:

```sh
==> Implied `brew uninstall --cask gimp`
==> Backing App 'GIMP.app' up to '/opt/homebrew/Caskroom/gimp/2.10.38,1/GIMP.app'
==> Removing App '/Applications/GIMP.app'
==> Unlinking Binary '/opt/homebrew/bin/gimp'
==> Dispatching zap stanza
==> Trashing files:
~/Library/Application Support/Gimp
~/Library/Preferences/org.gimp.gimp-2.10.plist
~/Library/Saved Application State/org.gimp.gimp-2.10.savedState
==> Removing all staged versions of Cask 'gimp'
-rw-------@ 1 daniel  staff  126 Jul 22  2023 /Users/daniel/Library/Preferences/org.gimp.gimp-2.10.plist
```

**Reply by SoCuul** [reply]
@SMillerDev if there's any log files I could provide or tests I could run, I'm more than happy to

**Comment by gromgit** [comment]
It seems [another user ran into a similar issue](https://github.com/orgs/Homebrew/discussions/5912), which seemed to be related to a [general CLT-related issue](https://github.com/orgs/Homebrew/discussions/5723). Can you try manually removing and reinstalling the CLT, as @Bo98 described in the second link above, then see if that fixes your problem like for @karmeye?
